driver-core/src/test/java/com/datastax/driver/core/WarningsTest.java

```diff
@@ -50,4 +54,52 @@ public void should_expose_warnings_on_execution_info() {
     List<String> warnings = rs.getExecutionInfo().getWarnings();
     assertThat(warnings).hasSize(1);
   }
+
+  @Test(groups = "short")
+  public void should_execute_query_with_server_side_warnings() {
+    // Set the system property to enable logging of server side warnings
+    final String originalLoggingFlag =
+        System.getProperty(RequestHandler.LOG_REQUEST_WARNINGS_PROPERTY, "false");
+    System.setProperty(RequestHandler.LOG_REQUEST_WARNINGS_PROPERTY, "true");
+    assertThat(Boolean.getBoolean(RequestHandler.LOG_REQUEST_WARNINGS_PROPERTY)).isTrue();
+    // create a TestAppender and add it
+    TestAppender logAppender = new TestAppender();
+    Logger.getRootLogger().addAppender(logAppender);
+    // Given a query that will produce server side warnings that will be embedded in the
+    // ExecutionInfo
+    SimpleStatement statement = new SimpleStatement("SELECT count(*) FROM foo");
+
+    ResultSet rs = session().execute(statement);
+    ExecutionInfo ei = rs.getExecutionInfo();
+    // When
+    Row row = rs.one();
+
+    // Then
+    assertThat(row).isNotNull();
+    List<String> warnings = ei.getWarnings();
+
+    assertThat(warnings).isNotEmpty();
+    assertThat(warnings.size()).isEqualTo(1);
+    assertThat(warnings.get(0)).isEqualTo("Aggregation query used without partition key");
+    // assert the log was generated
+    assertThat(logAppender).isNotNull();
+    assertThat(logAppender.logs).isNotNull();
+    assertThat(logAppender.logs).isNotEmpty();
+    assertThat(logAppender.logs.get(0).getMessage())
+        .isEqualTo("Aggregation query used without partition key");
+    // remove the log appender
+    Logger.getRootLogger().removeAppender(logAppender);
+    // reset the logging flag
+    System.setProperty(RequestHandler.LOG_REQUEST_WARNINGS_PROPERTY, originalLoggingFlag);
+  }
+
+  private class TestAppender extends WriterAppender {
```

Comment:
We already have a `MemoryAppender` class that works this way, and handles a couple of corner cases we ran into over time. I think you could use it here.